control_protocol.md

```diff
@@ -330,17 +330,490 @@ A Component MUST also offer a list of all possibly callable methods in accordanc
+                },
+                "required": true
+            }
+        }
+    ],
+    "components": {
+        "schemas": {}
+    }
+}
+:::
+
+Any Component MAY offer ANY of the following methods.
+Components SHOULD offer ``shut_down``.
+
+::: json
+{
+    "openrpc": "1.2.6",
+    "info": {
+        "title": "Component-optional",
+        "version": "0.1.0"
+    },
+    "methods": [
+        {
+            "name": "set_log_level",
+            "params": [
+                {
+                    "name": "level",
+                    "schema": {
+                        "type": "integer"
+                    },
+                    "required": true
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "shut_down",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+    ],
+    "components": {
+        "schemas": {}
+    }
+}
+:::
+
+
+#### Coordinator
+
+Control protocol Coordinators are also {ref}`Components <control_protocol.md#Component>` and MUST offer methods accordingly.
+Furthermore, Coordinators MUST offer the following methods.
+
+::: json
+{
+    "openrpc": "1.2.6",
+    "info": {
+        "title": "Coordinator",
+        "version": "0.1.0"
+    },
+    "methods": [
+        {
+            "name": "sign_in",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "sign_out",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "coordinator_sign_in",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "coordinator_sign_out",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "set_nodes",
+            "params": [
+                {
+                    "name": "nodes",
+                    "schema": {
+                        "type": "object",
+                        "additionalProperties": true
```

Comment:
Not strings with node names? Isn't it weird to send whole node objects over the wire?
IMO, if possible we should stay text-based in our message interface, no?

Reply:
I called the "rpc.discover" methods on the different devices to get their (openrpc) response. From these responses I extracted the methods.
In this case, I might have not defined the proper parameter type, such that the rpc server did not know it better.

Reply:
Ah, so this is WIP and can be made more specific (from type hints or so)?

Reply:
I just looked at the python file `leco_protocols`: I defined `dict` without any other declaration. Here we have the consequences.

Reply:
Yes, it "relies heavily" (quote from the package) on type hints.

I reworked all the json files according to open-rpc specs and json-schema.